### ISSUE TITLE
Add Dockerfile

### ISSUE THREAD

**Issue by navdeep-G** [h2oai/rsparkling#56]
* Would be good to have a Dockerfile for each type of h2o version + sparkling water version (latest for spark 2.0,2.1, and 2.2 to start with).

**Comment by navdeep-G** [comment]
@javierluraschi Would this help with installs of rsparkling? i.e., a user can run a container with specified environments.

**Comment by javierluraschi** [comment]
@navdeep-G I think it would!

Short term, I don't think `docker` is yet a ubiquitous tool in the data science community, but is getting there. Until recently, support for OS X / Windows was limited. There seems to be `docker` package in CRAN but depends on `python` through `reticulate`. So my guess is that installing `rsparkling` might be harder since most users in the R community would have to install `docker` + `python`. However, it is often the case that most advanced users of `sparklyr` are the ones that inquiry about H2O; therefore, there is a good chance that those users will already be familiar with docker and would find these images very helpful.

Long term, Docker is obviously here to stay and is becoming more and more common in the R community, with better support for OS X and Windows, and if someone were to provide a magic `docker` package that performs all the installation and configuration where we could run with ease something like:

```r
library(docker)
library(rsparklyring)
library(h2o)
library(sparklyr)

docker_install()
docker_start(rsparkling_docker_file())

spark_install()
sc <- spark_connect(master = rsparkling_docker_master())
```

it would obviously be really compelling.

In the meantime, it would help to have a mapping of Docker images in `rsparkling`, as in `rsparkling_docker_file()`, that lets users get a file by version and get the master `rsparkling_docker_master()`, or something along those lines.

**Comment by navdeep-G** [comment]
Yes, a package that handles configuration would be ideal. Maybe this is along those lines? https://github.com/rocker-org/rocker. I truly believe Docker is going to be big for a long time. I will brainstorm this some more, but your code snippet provides me with a sensible API.

**Comment by pgensler** [comment]
I would be extremely interested in having an docker image to pull from to use H20 in R, as I know it's not always the easiest to simply setup, and get up and running.

In case you are interested, the following docker container has RStudio along with Sparklyr, and H20 prebuilt. Maybe making a spinoff of this container is what you are after? You can easily add more install statements to the docker file, which is shown on the below, making installing additional packages a breeze like so:
https://hub.docker.com/r/methodsconsultants/tidyverse-h2o/

```
docker run -d -p 8787:8787 -v ~/Desktop:/home/rstudio methodsconsultants/tidyverse-h2o
```

This binds your Desktop as the /home/rstudio path. Note that RStudio needs port 8787 to run on.
